### PR TITLE
Update to user/developer documentation for finalizing 3.0 release

### PR DESCRIPTION
**Pull request by jjwilke**

### CODE REVIEW

**Comment by jjwilke** [comment]
I would keep this open for doc fixes and wait to merge until the very end.

File: BUILD.md [hunk 1]
```diff
@@ -156,9 +150,9 @@ Options can be enabled by specifiying `-DKokkos_ENABLE_X`.
     * BOOL Default: OFF
 * Kokkos_ENABLE_DEPRECATED_CODE
     * Whether to enable deprecated code
-    * BOOL Default: OFF
+    * BOOL Default: ON
```
Comment (dalg24) [inline BUILD.md]: ??????????

File: BUILD.md [hunk 2]
```diff
@@ -142,8 +136,8 @@ Options can be enabled by specifiying `-DKokkos_ENABLE_X`.
 * Kokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE
     * Whether to enable relocatable device code (RDC) for CUDA
     * BOOL Default: OFF
-* Kokkos_ENABLE_CUDA_UVM
-    * Whether to enable unified virtual memory (UVM) for CUDA
+* Kokkos_ENABLE_CUDA_DEFAULT_UM
+    * Whether to use unified memory (UM) by default for CUDA
```
Comment (dalg24) [inline BUILD.md]: Yes to update the description, no to rename.  I suppose this is related to #2289 
This needs more discussion and belong to a separate PR.

File: BUILD.md [hunk 3]
```diff
@@ -114,12 +114,6 @@ Device backends can be enabled by specifiying `-DKokkos_ENABLE_X`.
 * Kokkos_ENABLE_PTHREAD
     * Whether to build Pthread backend
     * BOOL Default: OFF
-* Kokkos_ENABLE_QTHREAD
-    * Whether to enable the QTHREAD library
-    * BOOL Default: OFF
```
Comment (masterleinad) [inline BUILD.md]: Do we actually want to remove the `QTHREAD` and `ROCM` backend completely for the release or just deprecate them?

File: BUILD.md (hunk 1 above)
Comment (crtrott) [reply BUILD.md]: Nope, default is now off.

File: BUILD.md (hunk 2 above)
Comment (crtrott) [reply BUILD.md]: Yeah we need to discuss this in the meeting.

**Comment by dalg24** [comment]
Closing.  Will be addressed soon in other PRs.
